### PR TITLE
Improve senza/utils.py based on pylint

### PR DESCRIPTION
**Pull request by jmcs**
- Docstrings;
- Argument name changed in `named_value` (still not really better);
- Removed one step from `camel_case_to_underscore` (@hjacobs I couldn't find any case that failed without it but maybe I'm missing something can you check this change?)

### CODE REVIEW

File: senza/utils.py [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+    parameters or Senza Components)
+    """
+    return next(iter(dictionary.items()))
 
 
 def ensure_keys(dict_obj, *keys):
+    """
+    Ensure ``dict_obj`` has the hierarchy ``{keys[0]: {keys[1]: {...}}}``
+
+    The inner most key will have ``{}`` has value if didn't exist already.
```
Comment (rafaelcaricio) [inline senza/utils.py]: innermost

**Comment by rafaelcaricio** [comment]
👍

**Comment by hjacobs** [comment]
@jmcs your change will fail for "ThisIsABook" (now it won't generate an underscore between "a" and "book"). So I would say it's better to keep the extra step.

**Comment by jmcs** [comment]
@hjacobs I re-added the extra step and added some unit tests with strings with more than one uppercase in a row.

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**